draft-ietf-taps-interface.md

```diff
@@ -258,6 +258,169 @@ Errors and other notifications also happen asynchronously on the Connection.
 through Actions and Events in each phase of a Connection, following the phases
 described in {{I-D.ietf-taps-arch}}.
 
+## Usage Examples
+
+The following usage examples illustrate how an application might use a
+Transport Services Interface to
+
+- Act as a server: Listen for incoming connections, receive a request, and send
+  a response, see {{server-example}}.
+- Act as a client: Connect to a remote endpoint, send a request, and receive a
+  response, see {{client-example}}.
+- Act as a peer: Connect to a remote endpoint using Rendezvous, send a Message,
+  and receive a Message, see {{peer-example}}.
+
+The examples in this section presume that a transport protocol is available
+between the endpoints which provides Reliable Data Transfer, Preservation of
+data ordering, and Preservation of Message Boundaries. In this case, the
+application can choose to receive only complete messages.
+
+If none of the available transport protocols provides Preservation of Message
+Boundaries, but there is a transport protocol which provides a reliable ordered
+octet stream, an application may receive this octet stream as partial
+Messages and transform it into application-layer Messages.  Alternatively,
+an application may provide a Deframer, which is a function that transforms an
+octet stream into a sequence of Messages, see {{receive-framing}}.
+
+
+### Server Example
+
+This is an example of how an application might listen for incoming Connections
+using the Transport Services Interface, receive a request, and send a response.
+
+~~~
+LocalSpecifier := NewLocalEndpoint()
+LocalSpecifier.WithInterface("en0")
+LocalSpecifier.WithService("https")
+
+TransportProperties := NewTransportProperties()
+TransportProperties.Require(reliability)
+TransportProperties.Require(preserve-order)
+TransportProperties.Require(preserve-msg-boundaries)
```

Comment:
I'd actually argue that we should not show setting these values. If we do believe that these are the defaults:

```
The recommended default is to Require Reliable Data Transfer.
```

Then why would the application need to set these explicitly? That just adds to boiler-plate code bulk.

Reply:
True for reliability and preserve order -- I wanted to show them explicitly, but I can also do that in a comment, and that's probably better.
For preserve-msg-boundaries, the default is Prefer, so I do think I need to set it here, otherwise I think the example will be somewhat ambiguous.

Reply:
Hm, okay. I think keeping one example of a changed property is good, so this one is fine. May want to change it in the future, but we can see.

Reply:
I _think_ we're leaning toward underscore-, as opposed to dash-separation for property names? in any case I don't think this has been decided yet so can leave it for now...